crates/router/src/core/admin.rs

```diff
@@ -3311,23 +3309,261 @@ pub async fn create_and_insert_business_profile(
 }
 
-#[cfg(all(
-    any(feature = "v1", feature = "v2"),
-    not(any(feature = "merchant_account_v2", feature = "business_profile_v2"))
-))]
+#[cfg(feature = "olap")]
+#[async_trait::async_trait]
+trait BusinessProfileCreateBridge {
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "business_profile_v2")
+    ))]
+    async fn create_domain_model_from_request(
+        self,
+        state: &SessionState,
+        merchant_account: &domain::MerchantAccount,
+        key: &domain::MerchantKeyStore,
+    ) -> RouterResult<domain::BusinessProfile>;
+
+    #[cfg(all(feature = "v2", feature = "business_profile_v2"))]
+    async fn create_domain_model_from_request(
+        self,
+        state: &SessionState,
+        key: &domain::MerchantKeyStore,
+        merchant_id: &id_type::MerchantId,
+    ) -> RouterResult<domain::BusinessProfile>;
+}
+
+#[cfg(feature = "olap")]
+#[async_trait::async_trait]
+impl BusinessProfileCreateBridge for api::BusinessProfileCreate {
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "business_profile_v2")
+    ))]
+    async fn create_domain_model_from_request(
+        self,
+        state: &SessionState,
+        merchant_account: &domain::MerchantAccount,
+        key_store: &domain::MerchantKeyStore,
+    ) -> RouterResult<domain::BusinessProfile> {
+        use common_utils::ext_traits::AsyncExt;
+
+        if let Some(session_expiry) = &self.session_expiry {
+            helpers::validate_session_expiry(session_expiry.to_owned())?;
+        }
+
+        if let Some(intent_fulfillment_expiry) = &self.intent_fulfillment_time {
+            helpers::validate_intent_fulfillment_expiry(intent_fulfillment_expiry.to_owned())?;
+        }
+
+        if let Some(ref routing_algorithm) = self.routing_algorithm {
+            let _: api_models::routing::RoutingAlgorithm = routing_algorithm
+                .clone()
+                .parse_value("RoutingAlgorithm")
+                .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                    field_name: "routing_algorithm",
+                })
+                .attach_printable("Invalid routing algorithm given")?;
+        }
+
+        // Generate a unique profile id
+        let profile_id = common_utils::generate_id_with_default_len("pro");
+        let profile_name = self.profile_name.unwrap_or("default".to_string());
+
+        let current_time = date_time::now();
+
+        let webhook_details = self.webhook_details.map(ForeignInto::foreign_into);
+
+        let payment_response_hash_key = self
+            .payment_response_hash_key
+            .or(merchant_account.payment_response_hash_key.clone())
+            .unwrap_or(common_utils::crypto::generate_cryptographically_secure_random_string(64));
+
+        let payment_link_config_value = self.payment_link_config.map(ForeignInto::foreign_into);
```

Comment:
Nit: This is no longer a `serde_json::Value`, I believe this is a remnant from #5531.

```suggestion
        let payment_link_config = self.payment_link_config.map(ForeignInto::foreign_into);
```